packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -345,6 +354,10 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
       validateName(props.canaryName);
     }
 
+    if (props.cleanup === Cleanup.LAMBDA && props.provisionedResourceCleanup) {
+      throw new Error('Cannot specify `provisionedResourceCleanup` when `cleanup` is set to `Cleanup.LAMBDA`. Use only `provisionedResourceCleanup`.');
```

Comment:
@badmintoncryer  I think the build is failing due to this error: `aws-cdk-lib: /codebuild/output/src1859401526/src/github.com/aws/aws-cdk/packages/aws-cdk-lib/aws-synthetics/lib/canary.ts
aws-cdk-lib:   358:13  error  Expected a non-default error object to be thrown  @cdklabs/no-throw-default-error`

I think the error needs to be updated from `Error` to `ValidationError` due to this PR where we are now enforcing specific error types: https://github.com/aws/aws-cdk/pull/33079